src/Control/Optics/Linear/Internal.hs

```diff
@@ -6,31 +6,442 @@
+--    includes the standard one
+--  * The instance marked by (4) implies that the linear iso definition
+--    includes the standard one
+--
+--
+-- == Examples
+--
+-- === 'Lens' Example
+--
+-- > {-# LANGUAGE LinearTypes #-}
+-- > {-# LANGUAGE StandaloneDeriving #-}
+-- > {-# LANGUAGE DeriveAnyClass #-}
+-- > {-# LANGUAGE TypeOperators #-}
+-- > {-# LANGUAGE FlexibleContexts #-}
+-- > {-# LANGUAGE RankNTypes #-}
+-- > {-# LANGUAGE GADTs #-}
+-- > {-# LANGUAGE RebindableSyntax #-}
+-- > {-# LANGUAGE RecordWildCards #-}
+-- > {-# LANGUAGE TypeApplications #-}
+-- > {-# LANGUAGE TypeInType #-}
+-- > {-# LANGUAGE TypeFamilies #-}
+-- > import qualified Data.Profunctor.Kleisli.Linear as Linear
+-- > import qualified Data.Profunctor.Linear as Linear
+-- > import Data.Bifunctor.Linear
+-- > import Control.Optics.Linear.Internal
+-- >
+-- > data Person where
+-- >   Person ::
+-- >     { age :: Int
+-- >     , name :: String
+-- >     , home :: Address
+-- >     } #-> Person
+-- >
+-- > data Address where
+-- >   Address ::
+-- >     { street :: String
+-- >     , zipcode :: Int
+-- >     } #-> Address
+-- >
+-- > personAddL :: Lens' Address Person
+-- > personAddL = lens (\(Person a n h) -> (h, Person a n))
+-- >
+-- > lens :: (s #-> (a, b #-> t)) -> Lens a b s t
+-- > lens decompose = Optical $ \a_arr_b ->
+-- >   Linear.dimap decompose apply (Linear.first a_arr_b)
+-- >   where
+-- >     apply :: (b, b #-> t) #-> t
+-- >     apply (b, f) = f b
+-- > overAddr :: (Address #-> Address) -> Person #-> Person
+-- > overAddr = over personAddL
+--
+-- === 'Traversal' Example
+--
+-- > {-# LANGUAGE LinearTypes #-}
+-- > {-# LANGUAGE StandaloneDeriving #-}
+-- > {-# LANGUAGE DeriveAnyClass #-}
+-- > {-# LANGUAGE TypeOperators #-}
+-- > {-# LANGUAGE FlexibleContexts #-}
+-- > {-# LANGUAGE RankNTypes #-}
+-- > {-# LANGUAGE GADTs #-}
+-- > {-# LANGUAGE RebindableSyntax #-}
+-- > {-# LANGUAGE RecordWildCards #-}
+-- > {-# LANGUAGE TypeApplications #-}
+-- > {-# LANGUAGE TypeInType #-}
+-- > {-# LANGUAGE TypeFamilies #-}
+-- > import qualified Data.Profunctor.Kleisli.Linear as Linear
+-- > import qualified Data.Profunctor.Linear as Linear
+-- > import Data.Bifunctor.Linear
+-- > import Control.Optics.Linear.Internal
+-- > import qualified Control.Monad.Linear as Control
+-- > import Data.Functor.Identity
+-- >
+-- > data IntStructure where
+-- >   OneInt :: Int #-> IntStructure
+-- >   SomeInts :: [Int] -> IntStructure
+-- >   deriving Show
+-- >
+-- > extract :: IntStructure #-> Maybe Int
+-- > extract (OneInt i) = Just i
+-- > extract (SomeInts []) = Nothing
+-- > extract (SomeInts (i:_)) = Just i
+-- >
+-- > buildUp :: Maybe Int #-> IntStructure
+-- > buildUp Nothing = SomeInts []
+-- > buildUp (Just i) = OneInt i
+-- >
+-- > -- A traversal that only looks at one int, in two different cases
+-- >
+-- > intStructT :: Traversal' Int IntStructure
+-- > intStructT = Optical $ \int_arr_int ->
+-- >   Linear.dimap extract buildUp (Linear.wander int_arr_int)
```

Comment:
Is this a lawful traversal? I might be wrong here, but `extract` loses some elements from the `IntStructure` which I find a bit strange. The `lens` documentation states that [http://hackage.haskell.org/package/lens-4.19.2/docs/Control-Lens-Traversal.html](page):

> a Traversal needs to leave the same number of elements as a candidate for subsequent Traversal that it started with.